docs/framework/unmanaged-api/profiling/icorprofilerinfo9-getcodeinfo4-method.md

```diff
@@ -0,0 +1,66 @@
+---
+title: "ICorProfilerInfo9::GetCodeInfo4"
+ms.date: "08/06/2019"
+dev_langs: 
+  - "cpp"
+api_name: 
+  - "ICorProfilerInfo9.GetCodeInfo4"
+api_location: 
+  - "mscorwks.dll"
+api_type: 
+  - "COM"
+ms.assetid: 4cd59947-7b29-4e6c-8866-725ec902bcc1
+author: "davmason"
+ms.author: "davmason"
+---
+# ICorProfilerInfo9::GetCodeInfo4 Method
+  
+ Given the native code start address, return the the blocks of virtual memory that store this code.
+  
+## Syntax  
+  
+```cpp
+HRESULT GetCodeInfo4( [in]  UINT_PTR pNativeCodeStartAddress,
+                      [in]  ULONG32 cCodeInfos,
+                      [out] ULONG32* pcCodeInfos,
+                      [out] COR_PRF_CODE_INFO codeInfos[]);
+```  
+  
+#### Parameters  
+ `pNativeCodeStartAddress`  
+ [in] A pointer to the start of a native function.  
+
+ `cCodeInfos`  
+ [in] The size of the `codeInfos` array.  
+  
+ `pcCodeInfos`  
+ [out] A pointer to the total number of [COR_PRF_CODE_INFO](../../../../docs/framework/unmanaged-api/profiling/cor-prf-code-info-structure.md) structures available.  
+  
+ `codeInfos`  
+ [out] A caller-provided buffer. After the method returns, it contains an array of `COR_PRF_CODE_INFO` structures, each of which describes a block of native code.  
+  
+## Remarks  
+ The `GetCodeInfo4` method is similar to [GetCodeInfo3](../../../../docs/framework/unmanaged-api/profiling/icorprofilerinfo4-getcodeinfo3-method.md), except that it can look up code info for different native versions of a method.  
```

Comment:
```suggestion
 The `GetCodeInfo4` method is similar to [GetCodeInfo3](../../../../docs/framework/unmanaged-api/profiling/icorprofilerinfo4-getcodeinfo3-method.md), except that it can look up code information for different native versions of a method.  
```